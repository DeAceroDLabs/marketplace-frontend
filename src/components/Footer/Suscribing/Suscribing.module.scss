@use "styles/colors";
@use 'styles/typography';

.subscribing {
  height: 100px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  padding: 00px 20px 20px;

  .text {
    padding: 20px;
    align-items: center;
    color: colors.$color-background-01;
  }

  .email {
    width: 55%;
    display: flex;
    justify-content: space-evenly;

    input {
      width: 40%;
      max-width: 289px;
      background-color: colors.$color-background-01;
      border: none;
      height: 42px;
      padding: 10px;
      border-radius: 8px;
    }

    .button-container {
      width: 50%;
    }

  }

}