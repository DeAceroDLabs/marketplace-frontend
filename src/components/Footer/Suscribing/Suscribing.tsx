import { useNavigate } from "react-router-dom";
import Button from "components/common/Button";
import styles from "./Suscribing.module.scss";

const Suscribing: React.FC = () => {
  const navigate = useNavigate();

  return (
    <div className={styles.subscribing}>
      <div className={styles.text}>
        Sé el primero en enterarte de ofertas y novedades. Compártenos tu correo
        y te mantendremos al tanto de todo.
      </div>
      <div className={styles.email}>
        <input placeholder="Escribe tu correo electónico" type="text" />
        <div className={styles["button-container"]}>
          <Button color="white" action={() => navigate("/")}>
            Suscribete
          </Button>
        </div>
      </div>
    </div>
  );
};

export default Suscribing;
