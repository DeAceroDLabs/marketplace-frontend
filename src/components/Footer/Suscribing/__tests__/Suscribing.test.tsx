import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Suscribing from "../Suscribing";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Suscribing", () => {
  it("renders Suscribing without problem", async () => {
    const view = render(
      <MemoryRouter>
        <Suscribing />
      </MemoryRouter>
    );
    expect(view).toMatchSnapshot();
  });
});
