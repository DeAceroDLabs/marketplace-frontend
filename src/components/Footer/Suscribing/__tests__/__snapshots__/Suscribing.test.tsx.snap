// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suscribing renders Suscribing without problem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="subscribing"
      >
        <div
          class="text"
        >
          Sé el primero en enterarte de ofertas y novedades. Compártenos tu correo y te mantendremos al tanto de todo.
        </div>
        <div
          class="email"
        >
          <input
            placeholder="Escribe tu correo electónico"
            type="text"
          />
          <div
            class="button-container"
          >
            <button
              class="button suscribe"
            >
              Suscribete
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="subscribing"
    >
      <div
        class="text"
      >
        Sé el primero en enterarte de ofertas y novedades. Compártenos tu correo y te mantendremos al tanto de todo.
      </div>
      <div
        class="email"
      >
        <input
          placeholder="Escribe tu correo electónico"
          type="text"
        />
        <div
          class="button-container"
        >
          <button
            class="button suscribe"
          >
            Suscribete
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
